### PR TITLE
Issue 99: meter action marking generalizing for all vendors

### PR DESCRIPTION
**Pull request by aseemavi**
meter action marking generalizing for all vendors

### CODE REVIEW

File: src/yang/ietf-qos-action.yang [hunk 1]
```diff
@@ -71,24 +71,31 @@ module ietf-qos-action {
...
       description
-        "Configuration of basic-meter & associated actions";
-
-      leaf action-type {
-        type identityref {
-          base qos-types:meter-action-type;
-        }
-        description
-          "meter action type";
+        "meter action type";
```
Comment (mjethanandani) [inline src/yang/ietf-qos-action.yang:79]: Possibly

"Meter action type."

File: src/yang/ietf-qos-action.yang [hunk 2]
```diff
@@ -71,24 +71,31 @@ module ietf-qos-action {
...
+    }
+    leaf dscp-mark {
+      type inet:dscp;
+      must "../action-type = 'qos-types:action-mark'" {
+        error-message
+        "The dscp must be marked when action-type is action-mark";
```
Comment (mjethanandani) [inline src/yang/ietf-qos-action.yang:85]: Indentation??

File: src/yang/ietf-qos-action.yang [hunk 3]
```diff
@@ -71,24 +71,31 @@ module ietf-qos-action {
...
       }
-
-      choice action-val {
-        description
-          "Meter action based on choice of meter action type";
+      description
+        "dscp marking";
```
Comment (mjethanandani) [inline src/yang/ietf-qos-action.yang:88]: "DSCP marking."

File: src/yang/ietf-qos-action.yang [hunk 4]
```diff
@@ -71,24 +71,31 @@ module ietf-qos-action {
...
+    }
+    leaf traffic-group-mark {
+      type string;
+      must "../action-type = 'qos-types:action-mark'" {
+        error-message
+        "The traffic-group must be marked when action-type is
```
Comment (mjethanandani) [inline src/yang/ietf-qos-action.yang:94]: Indentation??

File: src/yang/ietf-qos-action.yang [hunk 5]
```diff
@@ -71,24 +71,31 @@ module ietf-qos-action {
...
+         action-mark";
       }
+      description
+        "traffic group marking";
```
Comment (mjethanandani) [inline src/yang/ietf-qos-action.yang:98]: "Traffic group marking."

**Review by mjethanandani — CHANGES_REQUESTED** [review]

File: src/yang/ietf-qos-action.yang [hunk 6]
```diff
@@ -70,26 +70,33 @@ module ietf-qos-action {
...
+    }
+    leaf traffic-group-mark {
+      type string;
+      must "../action-type = 'qos-types:action-mark'" {
+        error-message
+          "Traffic-group must be marked when action-type is
+           action-mark";
       }
+      description
+        "Traffic group marking";
```
Comment (mjethanandani) [inline src/yang/ietf-qos-action.yang:99]: The type is a string, so it must be the name of the traffic-group. The description does not reflect the description of the leaf. Something like:

"A unique name in the form of a string for the traffic group."

File: src/yang/ietf-qos-action.yang [hunk 7]
```diff
@@ -70,26 +70,33 @@ module ietf-qos-action {
...
       description
-        "Configuration of basic-meter & associated actions.";
-
-      leaf action-type {
-        type identityref {
-          base qos-types:meter-action-type;
-        }
-        description
-          "Action type for metering.";
+        "Action type for metering.";
```
Comment (mjethanandani) [inline src/yang/ietf-qos-action.yang:80]: Indentation is off.

File: src/yang/ietf-qos-action.yang (hunk 6 above)
Comment (aseemavi) [reply src/yang/ietf-qos-action.yang:99]: how about "A unique name to mark as a traffic group."

File: src/yang/ietf-qos-action.yang (hunk 7 above)
Comment (aseemavi) [reply src/yang/ietf-qos-action.yang:80]: indentation is fine

**Review by mjethanandani — APPROVED** [review]

**Pull request merged**